From: Alexander Larsson <alexl@redhat.com>
Date: Mon, 16 Dec 2019 14:57:04 +0100
Subject: Remove extraneous ostree_async_progress_finish()

These should only be called at the leaf level, because the first
time its called no more change events will ever be sent on that
progress, which is not right possibly in the middle of an operation.

Origin: https://github.com/flatpak/flatpak/pull/3307
---
 common/flatpak-dir.c | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/common/flatpak-dir.c b/common/flatpak-dir.c
index b7436be..0fd125d 100644
--- a/common/flatpak-dir.c
+++ b/common/flatpak-dir.c
@@ -5319,9 +5319,6 @@ flatpak_dir_mirror_oci (FlatpakDir          *self,
   res = flatpak_mirror_image_from_oci (dst_registry, registry, oci_repository, oci_digest, ref, oci_pull_progress_cb,
                                        progress, cancellable, error);
 
-  if (progress)
-    ostree_async_progress_finish (progress);
-
   if (!res)
     return FALSE;
 
@@ -5410,9 +5407,6 @@ flatpak_dir_pull_oci (FlatpakDir          *self,
   checksum = flatpak_pull_from_oci (repo, registry, oci_repository, oci_digest, FLATPAK_OCI_MANIFEST (versioned), image_config,
                                     state->remote_name, ref, oci_pull_progress_cb, progress, cancellable, error);
 
-  if (progress)
-    ostree_async_progress_finish (progress);
-
   if (checksum == NULL)
     return FALSE;
 
@@ -5661,9 +5655,6 @@ out:
       g_assert (error == NULL || *error != NULL);
     }
 
-  if (progress)
-    ostree_async_progress_finish (progress);
-
   return ret;
 }
 
@@ -5746,9 +5737,6 @@ repo_pull_local_untrusted (FlatpakDir          *self,
   if (!res)
     translate_ostree_repo_pull_errors (error);
 
-  if (progress)
-    ostree_async_progress_finish (progress);
-
   return res;
 }
 
@@ -6008,9 +5996,6 @@ out:
   if (!ret)
     ostree_repo_abort_transaction (self->repo, cancellable, NULL);
 
-  if (progress)
-    ostree_async_progress_finish (progress);
-
   return ret;
 }
 
